src/nodes/chunk_append/chunk_append.c

```diff
@@ -310,21 +310,30 @@ ts_chunk_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht,
 			if (flat == NULL)
 				break;
 
+			/*
+			 * for each lc_oid, there will be 0, 1, or 2 matches in the flat list
+			 * if there are 2 matches they will also be consecutive (see assumption above)
+			 */
 			foreach (lc_oid, current_oids)
 			{
-				/* postgres may have pruned away some children already */
-				Path *child = (Path *) lfirst(flat);
-				Oid parent_relid = child->parent->relid;
-				bool is_not_pruned =
-					lfirst_oid(lc_oid) == root->simple_rte_array[parent_relid]->relid;
-
-				if (is_not_pruned)
+				bool matched = true;
+				do
 				{
-					merge_childs = lappend(merge_childs, child);
-					flat = lnext_compat(children, flat);
-					if (flat == NULL)
-						break;
-				}
+					Path *child = (Path *) lfirst(flat);
+					Oid parent_relid = child->parent->relid;
+					bool is_not_pruned =
```

Comment:
Do we need `is_not_pruned` and `matched`? Could this not be unified in one variable?

Reply:
you are right. Unless the child is pruned, there will be at least 1 match, so matched is redundant.